usr/bin/nxdk-cgc

```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env sh
+
+. ${NXDK_DIR}/usr/share/nxdk/prelude.inl
+
+if [ "$OS" = "macos" ]; then
+  CGC="${NXDK_DIR}/tools/cg/mac/cgc"
+elif [ "$OS" = "linux" ]; then
+  if [ "$UNAME_M" = "x86_64" ]; then
+    CGC="${NXDK_DIR}/tools/cg/linux/cgc"
+  elif [ "$UNAME_M" = "i686" ]; then
+    CGC="${NXDK_DIR}/tools/cg/linux/cgc.i386"
```

Comment:
Untested CPU detection

Reply:
You could consider dropping i686 support. i.e. macOS does no longer support running 32bit software at all.
Supporting less reduces the maintenance & testing overhead.

Reply:
i686 was specifically requested upstream.
This changeset also shouldn't try to do unrelated changes to nxdk.

All I did was to move these conditions from makefile to a `sh` script. If this version was removed from upstream nxdk then I'll remove it here.
Otherwise it will stay for compatibility sake and easier review (behaviour should match 1:1 to stock nxdk - aside from potential warning or log message changes).

Reply:
Agreed